roomserver/storage/postgres/storage.go

```diff
@@ -121,6 +134,27 @@ func (d *Database) StoreEvent(
 	}, nil
 }
 
+func extractRoomVersionFromCreateEvent(event gomatrixserverlib.Event) (
+	roomVersion gomatrixserverlib.RoomVersion, err error,
```

Comment:
Please don't use named return args here. I just spent five minutes writing a comment saying how the API shape is weird until I saw that `:144` assigns the return version to the default value. It would be clearer to explicitly return stuff here.

Reply:
Though the API shape is still a bit weird in that we return a zero-initialised version and *no error* if the event isn't a create event.